src/bot_export_guild_members_list.js

```diff
@@ -7,90 +7,155 @@ const client = new Client({
+    });
+  }
+}
 
-      if (process.env.DEBUG) {
-        // if in debug mode, just log the csvData to the console
-        console.log(csvData);
-        console.log("Finished.");
-        process.exit();
-      } else {
-        // if not in debug mode, write the csvData to a file for later
-        // processing
-        const formattedDate = new Date().toISOString(); // Format date as 'YYYY-MM-DDThh:mm:ssZ'
-        fs.writeFile(
-          `guild-${process.env.DISCORD_GUILD_ID}-members_list-${formattedDate}.csv`,
-          csvData,
-          (err) => {
-            if (err) throw err;
-            console.log("CSV file created!");
-            process.exit();
-          }
+/**
+ * Creates a CSV file for each Discord Role ID containing the Guild Members with
+ * that role.
+ * NOTES:
+ *   - If the role ID has no members, a CSV file is still created.
+ *   - If the role ID is not found, no CSV file is created.
+ *
+ * @param {Guild} guild is the Discord Guild object
+ * @param {string[]} roleIds is an array of Discord Role IDs from the .env file
+ */
+async function createCsvFilesForRoleIds(guild, roleIds) {
+  console.log("createCsvFilesForRoleIds() called...");
+  for (const roleId of roleIds) {
+    try {
+      console.log("\troleId:", roleId);
+      // NOTE: this should work but isn't, not sure if it's a permission
+      // issue, a bug, or I'm doing something wrong:
+      // guild.roles.cache.get(roleId)
+      //
+      // this successfully gets the role but not the role member data; see below
+      // `guild.members.fetch()` which seems to populate it just for this role.
+      // (so intuitive, right?)
+      const role = await guild.roles.fetch(roleId);
+      if (role) {
+        console.log(`\tRole: ${role.name}`);
+        // NOTE: `role.members` has no data unless `guild.members.fetch()` is
+        // called first:
+        await guild.members.fetch();
+        const members = role.members;
+        console.log("\tMembers with role:", members.size);
+        const csvData = getCsvDataForMembers(members);
+        await createCsvFile(
+          process.env.DISCORD_GUILD_ID,
+          roleId,
+          role.name,
+          csvData
         );
+      } else {
+        console.log("\tRole not found.");
       }
-    });
+    } catch (error) {
+      console.error("Error fetching members:", error);
+    }
+  }
+  console.log("...finished createCsvFilesForRoleIds()");
+}
+
+client.once(Events.ClientReady, async () => {
+  console.log(process.env.DEBUG ? "Debugging...\n" : "Creating CSV file...\n");
+  const guild = client.guilds.resolve(process.env.DISCORD_GUILD_ID);
+  await createCsvFilesForRoleIds(
+    guild,
+    process.env.DISCORD_ROLES.split(",")
+  ).then(() => {
+    if (process.env.DEBUG) {
+      console.log("Finished");
+    } else {
+      console.log("Files created");
+    }
+    // Can't close this as it'll close before the last role file finishes
+    // writing everything.
+    // process.exit();
```

Comment:
Can't figure out how to get the process to exit automatically once all roles are finished processing.